distage/distage-typesafe-config/src/main/scala/com/github/pshirshov/izumi/distage/config/typesafe/codec/RuntimeConfigReaderDefaultImpl.scala

```diff
@@ -59,7 +60,7 @@ class RuntimeConfigReaderDefaultImpl
 
   def deriveCaseClassReader(targetType: SafeType): ConfigReader[_] =
     cv => Try {
-      if(!symbolIntrospector.isConcrete(targetType)) {
+      if (!symbolIntrospector.isConcrete(targetType)) {
```

Comment:
copy methods from ReflectionProvider and SymbolIntrospector.

For `symbolIntrospector.selectConstructorMethod(targetType)`

use

```scala
import scala.reflect.runtime.universe._
val tpe: Type = ???

val constructor = tpe.decl(termNames.CONSTRUCTOR).asTerm.alternatives.head.asMethod

val parameters = constructor.typeSignatureIn(tpe).paramLists
```

Reply:
@ratoshniuk Done in https://github.com/pshirshov/izumi-r2/pull/372 please rebase